Sources/KituraNet/HTTP/HTTPServer.swift

```diff
@@ -309,13 +318,15 @@ public class HTTPServer: Server {
             }
             .childChannelInitializer { channel in
                 let httpHandler = HTTPRequestHandler(for: self)
+                let serverConfigurationHandler = HTTPServerConfigurationHandler(for: self)
                 let config: NIOHTTPServerUpgradeConfiguration = (upgraders: upgraders, completionHandler: { _ in
                     _ = channel.pipeline.removeHandler(httpHandler)
                 })
                 return channel.pipeline.configureHTTPServerPipeline(withServerUpgrade: config, withErrorHandling: true).flatMap {
                     if let nioSSLServerHandler = self.createNIOSSLServerHandler() {
                         _ = channel.pipeline.addHandler(nioSSLServerHandler, position: .first)
                     }
+                    _ = channel.pipeline.addHandler(serverConfigurationHandler, position: .first)
```

Comment:
all the lines that start with `_ = channel.pipeline` here aren't quite right. You're essentially racing the adding/removing of the handlers which isn't right. Basically you have two options:

1. chain the futures (using `channel.pipeline.addHandler(MyHandler()).flatMap { channel.pipeline.addHandler(MySecondHandler() } .flatMap { channel.pipeline.addHandler(MyThirdHandler() }`
2. prepare an array of handlers and add them in one shot using `addHandlers()`.

So here for example:

```swift
var allMyHandlers: [ChannelHandler] = []
if let nioSSLServerHandler = self.createNIOSSLServerHandler() {
    allMyHandlers.append(nioSSLServerHandler)
}
allMyHandlers.append(serverConfigurationHandler)
allMyHandlers.append(httpHandler)
return channel.pipeline.addHandlers(allMyHandlers)
```

Generally, whenever you need to use `_ = ` with SwiftNIO, you either found a bug (please report) or something went wrong. We tried fairly hard to mark all results that generally make sense to be ignored with `@discardableResult`. The result of `addHandler` is definitely not discardable.